include/f_core/device/c_gpio.h

```diff
@@ -0,0 +1,43 @@
+#ifndef C_GPIO_DEVICE
+#define C_GPIO_DEVICE
+
+#include <zephyr/drivers/gpio.h>
+
+class CGpio {
+public:
+    /**
+     * Constructor
+     * @param[in] dev Zephyr device structure
+     */
+    explicit CGpio(const gpio_dt_spec& gpioDev, const bool output = true) : gpioDev(&gpioDev) {
```

Comment:
Do we want the ability to pass in other gpio flags other than input (like activelow active high). Maybe we want a function that just passes through a zephyr gpio flags value to the pin

Reply:
Also or alternatively, we should probably have a way to get the underlying GPIO type such that if you need to interact with other zephyr APIs you can still use CGPio on top and only escpae when necessary